torch/jit/__init__.py

```diff
@@ -2079,6 +2079,80 @@ def _get_script_class(name):
                            "Did you forget to import it?".format(name))
     return _script_classes[name]
 
+# qualified_name => list[(overload decl, overload default args)]
+_overloaded_fns = {}
+
+# qualified name => list[compiled fns]
+_compiled_overloaded_fns = {}
+
+def overload(func):
+    qual_name = _qualified_name(func)
+    global _overloaded_fns
+    li = _overloaded_fns.get(qual_name)
+    if li is None:
+        li = []
+        _overloaded_fns[qual_name] = li
+    signature = torch.jit.annotations.get_signature(func)
+    if signature is None:
+        raise RuntimeError("Must explicitly add type annotations to overloaded functions: {obj}").format(func)
+    li.append((torch.jit.get_jit_def(func).decl(), get_default_args(func)))
+    return func
+
+def compile_function_with_overload(qual_name, impl_fn, overload_decl, overload_defaults):
+    _frames_up = 0
+    impl_ast = torch.jit.get_jit_def(impl_fn)
+    # TODO: figure out correct closure_rcb, stack_rcb, although this seems to work
```

Comment:
You probably need to leave this as is, see the comment in `_rcb` below, but we should at least avoid copy pasting this code to make a rcb from `script`